diesel/src/macros/mod.rs

```diff
@@ -228,108 +252,248 @@ macro_rules! __diesel_column {
 /// ```
 #[macro_export]
 macro_rules! table {
-    // Put `use` statements at the end because macro_rules! cannot figure out
-    // if `use` is an ident or not (hint: It's not)
+    // Put imports into the import field
     (
-        use $($import:tt)::+; $($rest:tt)+
+        @parse
+        import = [$(use $($import:tt)::+;)*];
+        doc = [$($doc:expr,)*];
+        use $($new_import:tt)::+; $($rest:tt)+
     ) => {
-        table!($($rest)+ use $($import)::+;);
+        table! {
+            @parse
+            import = [$(use $($import)::+;)* use $($new_import)::+;];
+            doc = [$($doc,)*];
+            $($rest)+
+        }
     };
-
-    // Add the primary key if it's not present
+    // Put doc annotation into the doc field
     (
-        $($table_name:ident).+ {$($body:tt)*}
-        $($imports:tt)*
+        @parse
+        import = [$(use $($import:tt)::+;)*];
+        doc = [$($doc:expr,)*];
+        #[doc=$new_doc:expr] $($rest:tt)+
```

Comment:
This is specifically the table level documentation, right? Is it even possible to have more than one of these?